sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -97,6 +97,42 @@ def test_direct_comment_after_docstring():
     assert result == expected_result
 
 
+def test_rst_block_after_docstring(gallery_conf):
+    """Assert there is a blank line between the docstring and rst blocks."""
+    with tempfile.NamedTemporaryFile('w', delete=False) as f:
```

Comment:
A way to do this more consistent with what we typically do (and that does not require try/except) is to use the `pytest` pattern:
```
def test_rst_block_after_docstring(gallery_conf, tmpdir):
    fname = os.path.join(tmpdir, 'temp.py')
    with open(fname, 'w') as fid:
         f.write(...)
    ...
```
And `pytest` will clean things up for you eventually.

Reply:
Took the other examples as a template. But you are right, tmpdir is simpler. Changed.